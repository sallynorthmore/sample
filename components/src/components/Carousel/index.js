import React from 'react';
import PropTypes from 'prop-types';
// import { IconChevron } from '../Icon';
import {
	Inner,
	Slides,
	Slide,
	SlideInner,
	Track,
	Navigation,
	NavButton,
	Progress,
	ProgressButton,
} from './styles';

class Carousel extends React.Component {
	state = {
		activeIndex: 0,
		renderedWidth: 0,
	};

	componentDidMount() {
		this.updateWidth();
		window.addEventListener('resize', this.updateWidth);
	}

	componentWillUnmount() {
		window.removeEventListener('resize', this.updateWidth);
	}

	onNavClick = direction => {
		const { children } = this.props;
		const { activeIndex } = this.state;

		this.setState({
			activeIndex:
				(activeIndex + children.length + direction) % children.length,
		});
	};

	goToSlide = index => {
		this.setState({
			activeIndex: index,
		});
	};

	updateWidth = () => {
		this.setState({
			renderedWidth: this.carouselDOM.offsetWidth,
		});
	};

	render() {
		const { children, width, responsive } = this.props;
		const { activeIndex, renderedWidth } = this.state;
		const carouselWidth = responsive ? '100%' : `${width}px`;
		const currentWidth = responsive ? renderedWidth : width;
		const slideWidth = currentWidth;
		const trackWidth = `${currentWidth * children.length}px`;
		const trackPosition = `-${currentWidth * activeIndex}px`;

		return (
			<Inner
				innerRef={elem => {
					this.carouselDOM = elem;
				}}
			>
				<Slides width={carouselWidth}>
					<Track width={trackWidth} position={trackPosition}>
						{children.map((item, index) => (
							<Slide
								key={index}
								width={slideWidth}
								active={activeIndex === index}
								position={slideWidth * index}
							>
								<SlideInner width={slideWidth}> {item}</SlideInner>
							</Slide>
						))}
					</Track>
				</Slides>
				<Navigation active={children.length > 1}>
					<NavButton onClick={() => this.onNavClick(-1)} previous>
						{/* <IconChevron width={24} /> */}
					</NavButton>
					<Progress>
						{children.map((item, index) => (
							<ProgressButton
								key={index}
								onClick={() => this.goToSlide(index)}
								active={activeIndex === index}
							/>
						))}
					</Progress>
					<NavButton onClick={() => this.onNavClick(1)}>
						{/* <IconChevron width={24} /> */}
					</NavButton>
				</Navigation>
			</Inner>
		);
	}
}

Carousel.defaultProps = {
	children: [],
	responsive: true,
	width: 540,
};

Carousel.propTypes = {
	children: PropTypes.array.isRequired,
	responsive: PropTypes.bool.isRequired,
	width: PropTypes.number,
};

export default Carousel;
