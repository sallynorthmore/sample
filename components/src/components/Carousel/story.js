import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { number, array, boolean, text } from '@storybook/addon-knobs';

import Carousel from './index';

const stories = storiesOf('Carousel', module);
const sampleItems = ['item 1', 'item 2', 'item 3'];

stories.add(
	'Carousel',
	withInfo('Carousel documentation')(() => {
		return (
			<div
				style={{
					backgroundColor: text('Background color', '#0b1a34'),
					minHeight: '500px',
					textAlign: 'center',
					display: 'flex',
				}}
			>
				<Carousel
					width={number('Carousel width in pixels', 540)}
					responsive={boolean('Responsive?', true)}
				>
					{array('Data array', sampleItems)}
				</Carousel>
			</div>
		);
	})
);
