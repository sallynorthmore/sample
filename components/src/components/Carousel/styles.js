import styled from 'styled-components';

const fade = `opacity .225s cubic-bezier(0.4, 0.0, 0.2, 1)`;

export const Inner = styled.div`
	align-items: center;
	color: white;
	display: flex;
	flex: 1;
	flex-direction: column;
	width: 100%;
`;

export const Slides = styled.div`
	display: flex;
	flex: 1;
	flex-direction: column;
	margin: 0;
	overflow: hidden;
	padding: 0;
	position: relative;
	transform: translate3d(0, 0, 0);
	width: ${props => props.width};
`;

export const Track = styled.div`
	display: flex;
	flex: 1;
	height: 100%;
	left: 0;
	position: relative;
	top: 0;
	/* stylelint-disable-next-line value-list-max-empty-lines */
	transform: ${props =>
		props.position
			? `translate3d(${props.position}, 0, 0)`
			: 'translate3d(0, 0, 0)'};
	transition: transform 0.225s cubic-bezier(0.4, 0, 0.6, 1);
	width: ${props => props.width};
`;

export const Slide = styled.div`
	/*  stylelint-disable order/properties-alphabetical-order */
	min-height: 1px;
	opacity: ${props => (props.active ? '1' : '0')};
	transition: ${fade};
	width: ${props => props.width}px;

	/* Chromium 49 hackhttps://github.com/philipwalton/flexbugs/issues/84 */
	bottom: 0;
	position: absolute;
	top: 0;
	display: block;
	left: ${props => props.position}px;
	/*  stylelint-enable */
`;

export const SlideInner = styled.div`
	height: 100%;
	width: ${props => props.width};
`;

export const Navigation = styled.div`
	align-items: center;
	border-top: 2px solid black;
	display: ${props => (props.active ? 'flex' : 'none')};
	flex: 0 0 64px;
	justify-content: space-between;
	padding: 24px 16px;
	width: 100%;
`;

export const NavButton = styled.button`
	align-items: center;
	background: transparent;
	border: 0 transparent;
	color: white;
	cursor: pointer;
	display: flex;
	flex: 0 0 40px;
	height: 40px;
	justify-content: center;
	opacity: 1;
	outline: none;
	transform: ${props => (props.previous ? 'rotate(90deg)' : 'rotate(-90deg)')};
	transition: ${fade};
	width: 40px;

	&:hover {
		opacity: 1;
	}
`;

export const Progress = styled.div`
	background: transparent;
	display: flex;
	flex: 1;
	justify-content: center;
	margin: 0 0 0 -8px;
`;

export const ProgressButton = styled.div`
	cursor: pointer;
	opacity: ${props => (props.active ? '1' : '0.3')};
	padding: 16px 8px;
	transition: ${fade};

	&::before {
		background: white;
		border-radius: 50%;
		content: '';
		display: block;
		height: 8px;
		width: 8px;
	}

	&:hover {
		opacity: 1;
	}
`;
