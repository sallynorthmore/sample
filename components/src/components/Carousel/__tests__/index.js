import React from 'react';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Carousel from '../index';

const testProps = {
	items: [''],
	firstSlide: 0,
	responsive: true,
	width: 540,
};

describe('<Carousel /> snapshot', () => {
	it('renders correctly', () => {
		const wrapper = mount(
			<Carousel
				firstSlide={testProps.firstSlide}
				responsive
				width={testProps.width}
			>
				{testProps.items}
			</Carousel>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders correctly without props', () => {
		const wrapper = mount(<Carousel />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
