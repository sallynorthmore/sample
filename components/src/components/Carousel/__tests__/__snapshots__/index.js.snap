// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> snapshot renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  overflow: hidden;
  padding: 0;
  position: relative;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  left: 0;
  position: relative;
  top: 0;
  -webkit-transform: translate3d(-0px,0,0);
  -ms-transform: translate3d(-0px,0,0);
  transform: translate3d(-0px,0,0);
  -webkit-transition: -webkit-transform 0.225s cubic-bezier(0.4,0,0.6,1);
  -webkit-transition: transform 0.225s cubic-bezier(0.4,0,0.6,1);
  transition: transform 0.225s cubic-bezier(0.4,0,0.6,1);
  width: 0px;
}

.c3 {
  min-height: 1px;
  opacity: 1;
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  width: 0px;
  bottom: 0;
  position: absolute;
  top: 0;
  display: block;
  left: 0px;
}

.c4 {
  height: 100%;
  width: 0;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 2px solid black;
  display: none;
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 24px 16px;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0 transparent;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  width: 40px;
}

.c6:hover {
  opacity: 1;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0 transparent;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  width: 40px;
}

.c9:hover {
  opacity: 1;
}

.c7 {
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 0 0 -8px;
}

.c8 {
  cursor: pointer;
  opacity: 1;
  padding: 16px 8px;
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
}

.c8::before {
  background: white;
  border-radius: 50%;
  content: '';
  display: block;
  height: 8px;
  width: 8px;
}

.c8:hover {
  opacity: 1;
}

<Carousel
  firstSlide={0}
  responsive={true}
  width={540}
>
  <styled.div
    innerRef={[Function]}
  >
    <div
      className="c0"
    >
      <styled.div
        width="100%"
      >
        <div
          className="c1"
          width="100%"
        >
          <styled.div
            position="-0px"
            width="0px"
          >
            <div
              className="c2"
              width="0px"
            >
              <styled.div
                active={true}
                key="0"
                position={0}
                width={0}
              >
                <div
                  className="c3"
                  width={0}
                >
                  <styled.div
                    width={0}
                  >
                    <div
                      className="c4"
                      width={0}
                    >
                       
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
      <styled.div
        active={false}
      >
        <div
          className="c5"
        >
          <styled.button
            onClick={[Function]}
            previous={true}
          >
            <button
              className="c6"
              onClick={[Function]}
            >
              <IconChevron
                fillColor="currentColor"
                percent={false}
                width={24}
              >
                <svg
                  viewBox="0 0 11 7"
                  width="24px"
                >
                  <g
                    fill="none"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      stroke="currentColor"
                      strokeWidth="1.2"
                      transform="translate(-321.000000, -147.000000)"
                    >
                      <polyline
                        points="324.045716 144.990072 329.062822 150.007178 324.009928 155.060072"
                        transform="translate(326.536375, 150.025072) rotate(90.000000) translate(-326.536375, -150.025072) "
                      />
                    </g>
                  </g>
                </svg>
              </IconChevron>
            </button>
          </styled.button>
          <styled.div>
            <div
              className="c7"
            >
              <styled.div
                active={true}
                key="0"
                onClick={[Function]}
              >
                <div
                  className="c8"
                  onClick={[Function]}
                />
              </styled.div>
            </div>
          </styled.div>
          <styled.button
            onClick={[Function]}
          >
            <button
              className="c9"
              onClick={[Function]}
            >
              <IconChevron
                fillColor="currentColor"
                percent={false}
                width={24}
              >
                <svg
                  viewBox="0 0 11 7"
                  width="24px"
                >
                  <g
                    fill="none"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      stroke="currentColor"
                      strokeWidth="1.2"
                      transform="translate(-321.000000, -147.000000)"
                    >
                      <polyline
                        points="324.045716 144.990072 329.062822 150.007178 324.009928 155.060072"
                        transform="translate(326.536375, 150.025072) rotate(90.000000) translate(-326.536375, -150.025072) "
                      />
                    </g>
                  </g>
                </svg>
              </IconChevron>
            </button>
          </styled.button>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Carousel>
`;

exports[`<Carousel /> snapshot renders correctly without props 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  overflow: hidden;
  padding: 0;
  position: relative;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  left: 0;
  position: relative;
  top: 0;
  -webkit-transform: translate3d(-0px,0,0);
  -ms-transform: translate3d(-0px,0,0);
  transform: translate3d(-0px,0,0);
  -webkit-transition: -webkit-transform 0.225s cubic-bezier(0.4,0,0.6,1);
  -webkit-transition: transform 0.225s cubic-bezier(0.4,0,0.6,1);
  transition: transform 0.225s cubic-bezier(0.4,0,0.6,1);
  width: 0px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 2px solid black;
  display: none;
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 24px 16px;
  width: 100%;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0 transparent;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  width: 40px;
}

.c4:hover {
  opacity: 1;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0 transparent;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  outline: none;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  transition: opacity .225s cubic-bezier(0.4,0.0,0.2,1);
  width: 40px;
}

.c6:hover {
  opacity: 1;
}

.c5 {
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 0 0 -8px;
}

<Carousel
  responsive={true}
  width={540}
>
  <styled.div
    innerRef={[Function]}
  >
    <div
      className="c0"
    >
      <styled.div
        width="100%"
      >
        <div
          className="c1"
          width="100%"
        >
          <styled.div
            position="-0px"
            width="0px"
          >
            <div
              className="c2"
              width="0px"
            />
          </styled.div>
        </div>
      </styled.div>
      <styled.div
        active={false}
      >
        <div
          className="c3"
        >
          <styled.button
            onClick={[Function]}
            previous={true}
          >
            <button
              className="c4"
              onClick={[Function]}
            >
              <IconChevron
                fillColor="currentColor"
                percent={false}
                width={24}
              >
                <svg
                  viewBox="0 0 11 7"
                  width="24px"
                >
                  <g
                    fill="none"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      stroke="currentColor"
                      strokeWidth="1.2"
                      transform="translate(-321.000000, -147.000000)"
                    >
                      <polyline
                        points="324.045716 144.990072 329.062822 150.007178 324.009928 155.060072"
                        transform="translate(326.536375, 150.025072) rotate(90.000000) translate(-326.536375, -150.025072) "
                      />
                    </g>
                  </g>
                </svg>
              </IconChevron>
            </button>
          </styled.button>
          <styled.div>
            <div
              className="c5"
            />
          </styled.div>
          <styled.button
            onClick={[Function]}
          >
            <button
              className="c6"
              onClick={[Function]}
            >
              <IconChevron
                fillColor="currentColor"
                percent={false}
                width={24}
              >
                <svg
                  viewBox="0 0 11 7"
                  width="24px"
                >
                  <g
                    fill="none"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      stroke="currentColor"
                      strokeWidth="1.2"
                      transform="translate(-321.000000, -147.000000)"
                    >
                      <polyline
                        points="324.045716 144.990072 329.062822 150.007178 324.009928 155.060072"
                        transform="translate(326.536375, 150.025072) rotate(90.000000) translate(-326.536375, -150.025072) "
                      />
                    </g>
                  </g>
                </svg>
              </IconChevron>
            </button>
          </styled.button>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Carousel>
`;
